### PR TITLE
Make it possible to intelligently compare headers

### PR DESCRIPTION
**Pull request by rominf**
To accomplish that HeaderValueMatcher was added. It already contains logic to compare unknown headers and authorization headers.

Tests were also added.

### CODE REVIEW

File: pytest_httpserver/httpserver.py [hunk 1]
```diff
@@ -51,14 +53,44 @@ class RequestMatcher:
     :param query_string: the http query string starting with ``?``, such as ``?username=user``
     """
 
+    class HeaderValueMatcher:
```
Comment (csernazs) [inline pytest_httpserver/httpserver.py]: Could you move this class to the module level? I think having class definition in a class is very unusual in python and having a separate class would make clearer code.

File: pytest_httpserver/httpserver.py [hunk 2]
```diff
@@ -51,14 +53,44 @@ class RequestMatcher:
...
+        @staticmethod
+        def authorization_header_value_matcher(actual: str, expected: str) -> bool:
+            return parse_authorization_header(actual) == parse_authorization_header(expected)
+
+        @staticmethod
+        def default_header_value_matcher(actual: str, expected: str) -> bool:
+            return actual == expected
+
+        def __call__(self, header_name: str, actual: str, expected: str) -> bool:
+            matcher = self.matchers[header_name]
```
Comment (csernazs) [inline pytest_httpserver/httpserver.py]: what would happen if self.matchers[header_name] raises a KeyError? Do you want to handle it or are you ok with raising a KeyError here? Just asking...

File: pytest_httpserver/httpserver.py (hunk 1 above)
Comment (rominf) [reply pytest_httpserver/httpserver.py]: Done

File: pytest_httpserver/httpserver.py (hunk 2 above)
Comment (rominf) [reply pytest_httpserver/httpserver.py]: Yep, you are right, I reraise custom exception.

**Comment by csernazs** [comment]
Thanks! :)

**Comment by rominf** [comment]
You are welcome!
Can you bump the version and upload package on PyPI?

**Comment by csernazs** [comment]
Yes, I'm on it. I'm writing the release notes entry based on your commit message, and will do the release.

**Comment by rominf** [comment]
Ok, thanks!
Please ping me so that I know when it's done.

**Comment by csernazs** [comment]
It has been reelased to pypi:
https://pypi.org/project/pytest-httpserver/#files

And the documentation is also updated, here is the release notes:
https://pytest-httpserver.readthedocs.io/en/latest/changes.html

I hope I didn't left anything out. :)

**Comment by rominf** [comment]
Thanks again! Your library saved us from ugly code handling requests manually in our tests.

**Pull request merged**